bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitAccessoryFactory.java

```diff
@@ -196,7 +194,7 @@ public static HomekitAccessory create(HomekitTaggedItem taggedItem, MetadataRegi
             }
         } catch (NoSuchMethodException | IllegalAccessException | InstantiationException
                 | InvocationTargetException e) {
-            logger.warn("Cannot instantiate accessory implementation for accessory {}", accessoryType.getTag(), e);
+            logger.warn("Cannot instantiate accessory implementation for accessory {}", accessoryType.getTag());
             throw new HomekitException("Cannot instantiate accessory implementation for accessory " + accessoryType);
```

Comment:
If you are going to remove the stack trace from the logs, then you should make sure that the exception is included as a cause exception when throwing.
```suggestion
            throw new HomekitException("Cannot instantiate accessory implementation for accessory " + accessoryType, e);
```

Reply:
you are right.. to much clean up. better to keep stack trace. reverted back the change